oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java

```diff
@@ -336,22 +336,42 @@ protected void indexSimilarityBinaries(Document doc, PropertyDefinition pd, Blob
             }
         }
     }
-    
+
     @Override
     protected boolean indexDynamicBoost(Document doc, PropertyDefinition pd, NodeState nodeState, String propertyName) {
         NodeState propertNode = nodeState;
         String parentName = PathUtils.getParentPath(propertyName);
         for (String c : PathUtils.elements(parentName)) {
-            propertNode = propertNode.getChildNode(c); 
+            propertNode = propertNode.getChildNode(c);
         }
         boolean added = false;
         for (String nodeName : propertNode.getChildNodeNames()) {
             NodeState dynaTag = propertNode.getChildNode(nodeName);
-            String dynaTagName = dynaTag.getProperty(DYNAMIC_BOOST_TAG_NAME).getValue(Type.STRING);
-            Double dynaTagConfidence = dynaTag.getProperty(DYNAMIC_BOOST_TAG_CONFIDENCE).getValue(Type.DOUBLE);
-
+            PropertyState p = dynaTag.getProperty(DYNAMIC_BOOST_TAG_NAME);
+            if (p == null || p.isArray()) {
+                continue;
+            }
+            String dynaTagName = p.getValue(Type.STRING);
+            if (dynaTagName == null) {
+                continue;
+            }
+            p = dynaTag.getProperty(DYNAMIC_BOOST_TAG_CONFIDENCE);
+            if (p == null || p.isArray()) {
+                continue;
+            }
+            Double dynaTagConfidence;
```

Comment:
minor: I would use `double` as primitive here since the object is unboxed to a primitive twice (line 369 and 381).

Reply:
Yes.